Sources/NIO/SocketAddresses.swift

```diff
@@ -308,3 +308,28 @@ extension SocketAddress: Equatable {
     }
 }
 
+
+extension SocketAddress {
+    /// Whether this `SocketAddress` corresponds to a multicast address.
+    public var isMulticast: Bool {
+        switch self {
+        case .unixDomainSocket:
+            // No multicast on unix sockets.
+            return false
+        case .v4(let v4Addr):
+            // For IPv4 a multicast address is in the range 224.0.0.0/4.
+            // The easy way to check if this is the case is to just mask off
+            // the address.
+            let v4WireAddress = v4Addr.address.sin_addr.s_addr
+            let mask = in_addr_t(bigEndian: 0xE000_0000)
+            return v4WireAddress & mask == mask
```

Comment:
this isn't quite right, is it? I think it should be

```swift
let mask = in_addr_t(bigEndian: 0xF000_0000)
let twoTwoFour = in_addr_t(bigEndian: 0xE000_000)
return v4WireAddress & mask == twoTwoFour
```

otherwise you're testing for `224.0.0.0/3`, no?

Reply:
also actually the `bigEndian` label is wrong as the data isn't necessarily in big endian, it should be `0xE000_000.bigEndian` same for the others. The other option is to turn the v4 wire address into host endianness using `in_addr_t(bigEndian: v4Addr.address.sin_addr.s_addr`. Either way works but not `in_addr_t(bigEndian: 0xE000_0000)`